src/ring_theory/witt_vector/mul_coeff.lean

```diff
@@ -0,0 +1,339 @@
+/-
+Copyright (c) 2022 Robert Y. Lewis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Robert Y. Lewis, Heather Macbeth
+-/
+
+import ring_theory.witt_vector.truncated
+import data.mv_polynomial.supported
+
+/-!
+# Leading terms of Witt vector multiplication
+
+The goal of this file is to study the leading terms of the formula for the `n+1`st coefficient
+of a product of Witt vectors `x` and `y` over a ring of characteristic `p`.
+We aim to isolate the `n+1`st coefficients of `x` and `y`, and express the rest of the product
+in terms of a function of the lower coefficients.
+
+For most of this file we work with terms of type `mv_polynomial (fin 2 × ℕ) ℤ`.
+We will eventually evaluate them in `k`, but first we must take care of a calculation
+that needs to happen in characteristic 0.
+
+## Main declarations
+
+* `witt_vector.nth_mul_coeff`
+
+-/
+
+noncomputable theory
+
+namespace witt_vector
+
+variables (p : ℕ) [hp : fact p.prime]
+variables {k : Type*} [comm_ring k]
+local notation `𝕎` := witt_vector p
+open finset mv_polynomial
+open_locale big_operators
+
+/--
+```
+(∑ i in range n, (y.coeff i)^(p^(n-i)) * p^i.val) *
+  (∑ i in range n, (y.coeff i)^(p^(n-i)) * p^i.val)
+```
+-/
+def witt_poly_prod (n : ℕ) : mv_polynomial (fin 2 × ℕ) ℤ :=
+rename (prod.mk (0 : fin 2)) (witt_polynomial p ℤ n) *
+  rename (prod.mk (1 : fin 2)) (witt_polynomial p ℤ n)
+
+include hp
+
+lemma witt_poly_prod_vars (n : ℕ) :
+  (witt_poly_prod p n).vars ⊆ finset.univ.product (finset.range (n + 1)) :=
+begin
+  rw [witt_poly_prod],
+  apply subset.trans (vars_mul _ _),
+  apply union_subset;
+  { apply subset.trans (vars_rename _ _),
+    simp [witt_polynomial_vars,image_subset_iff] }
+end
+
+private lemma sum_ident_1 (n : ℕ) :
+  (∑ i in range (n+1), p^i * (witt_mul p i)^(p^(n-i)) : mv_polynomial (fin 2 × ℕ) ℤ) =
+    witt_poly_prod p n :=
+begin
+  simp only [witt_poly_prod],
+  convert witt_structure_int_prop p (X (0 : fin 2) * X 1) n using 1,
+  { simp only [witt_polynomial, witt_mul, int.nat_cast_eq_coe_nat],
+    rw alg_hom.map_sum,
+    congr' 1 with i,
+    congr' 1,
+    have hsupp : (finsupp.single i (p ^ (n - i))).support = {i},
+    { rw finsupp.support_eq_singleton,
+      simp only [and_true, finsupp.single_eq_same, eq_self_iff_true, ne.def],
+      exact pow_ne_zero _ hp.out.ne_zero, },
+    simp only [bind₁_monomial, hsupp, int.cast_coe_nat, prod_singleton, ring_hom.eq_int_cast,
+      finsupp.single_eq_same, C_pow, mul_eq_mul_left_iff, true_or, eq_self_iff_true], },
+  { simp only [map_mul, bind₁_X_right] }
+end
+
+/-- The "remainder term" of `witt_poly_prod`. See `sum_ident_2`. -/
```

Comment:
To have a clickable link should we write `witt_vector.witt_poly_prod`? I don't know exactly how this works...

Reply:
```suggestion
/-- The "remainder term" of `witt_vector.witt_poly_prod`. See `sum_ident_2`. -/
```